common/src/collision.rs

```diff
@@ -294,59 +243,58 @@ mod tests {
     const CHUNK_SIZE: u8 = 12_u8; // might want to test with multiple values in the future.
     static CHUNK_SIZE_F: f64 = CHUNK_SIZE as f64;
 
-    // place a small bounding box near the center of the node. There should be exactly 20 chunks in contact.
-    #[test]
-    fn proper_chunks_20() {
+    fn proper_chunks_generic(x: f64, y: f64, z: f64, expected_chunks: usize) {
         let mut graph = Graph::new();
         ensure_nearby(&mut graph, &Position::origin(), 4.0);
-
-        let central_chunk = Vertex::A; // arbitrary vertex
-        let position = central_chunk.chunk_to_node()
-            * na::Vector4::new(
-                0.4 / CHUNK_SIZE_F,
-                0.4 / CHUNK_SIZE_F,
-                0.4 / CHUNK_SIZE_F,
-                1.0,
-            );
+        let central_chunk = Vertex::B; // arbitrary vertex
+        let position = central_chunk.chunk_to_node() * na::Vector4::new(x, y, z, 1.0);
 
         let bb = BoundingBox::create_aabb(
             NodeId::ROOT,
             position,
-            2.0 / CHUNK_SIZE_F,
+            1.6 / CHUNK_SIZE_F,
             &graph,
             CHUNK_SIZE,
         );
 
-        assert_eq!(bb.bounding_boxes.len(), 20);
+        assert_eq!(bb.bounding_boxes.len(), expected_chunks);
     }
 
-    // place a small bounding box in the center of a chunk. There should be exactly 1 chunk in contact.
+    // place a small bounding box near the center of the node. There should be exactly 20 chunks in contact.
     #[test]
-    fn proper_chunks_1() {
-        let mut graph = Graph::new();
-        ensure_nearby(&mut graph, &Position::origin(), 4.0);
-
-        let central_chunk = Vertex::A; // arbitrary vertex
-        let position = central_chunk.chunk_to_node() * na::Vector4::new(0.5, 0.5, 0.5, 1.0);
-
-        let bb = BoundingBox::create_aabb(
-            NodeId::ROOT,
-            position,
-            1.0 / CHUNK_SIZE_F,
-            &graph,
-            CHUNK_SIZE,
+    fn proper_chunks_20() {
+        proper_chunks_generic(
+            0.4 / CHUNK_SIZE_F,
+            0.4 / CHUNK_SIZE_F,
+            0.4 / CHUNK_SIZE_F,
+            20,
         );
+    }
 
-        assert_eq!(bb.bounding_boxes.len(), 1);
+    // place a small bounding box in the center of a chunk. There should be exactly 1 chunk in contact.
+    #[test]
+    fn proper_chunks_1() {
+        proper_chunks_generic(0.5, 0.5, 0.5, 1)
     }
 
     // place a small bounding box next to a dodecaherdral vertex. There should be exactly 8 chunks in contact.
     #[test]
     fn proper_chunks_8() {
```

Comment:
`PERPENDICULAR_VERTEX` is not yet included in the `proper_chunks_n` tests. There should be another test added to the suite. Its associated comment should be something like
```rust
// place a small bounding box near the midpoint of a dodecahedron's edge. There should be exactly 8 chunks in contact.
```
The chunk coordinates should be close to any permutation of (1, 1, 0).